src/main/java/slimeknights/tconstruct/smeltery/block/CastingTankBlock.java

```diff
@@ -0,0 +1,151 @@
+package slimeknights.tconstruct.smeltery.block;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.EntityBlock;
+import net.minecraft.world.level.block.Mirror;
+import net.minecraft.world.level.block.Rotation;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.DirectionProperty;
+import net.minecraft.world.level.material.PushReaction;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.HitResult;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.mantle.block.InventoryBlock;
+import slimeknights.mantle.util.BlockEntityHelper;
+import slimeknights.tconstruct.library.utils.NBTTags;
+import slimeknights.tconstruct.smeltery.block.entity.CastingTankBlockEntity;
+import slimeknights.tconstruct.smeltery.block.entity.CastingTankBlockEntity.ITankBlock;
+import slimeknights.tconstruct.smeltery.block.entity.ITankBlockEntity;
+
+import javax.annotation.Nullable;
+
+public class CastingTankBlock extends InventoryBlock implements ITankBlock, EntityBlock {
+  protected static final DirectionProperty FACING = BlockStateProperties.HORIZONTAL_FACING;
+  private final PushReaction pushReaction;
```

Comment:
The push reaction field is only used in tanks because we vary it. For this, just override the getter; no need for constructor parameter.

Though tbh, I have nothing against it being pushed, just let the vanilla default prevent it due to block entity (so mods like quark can still push it)